server/src/main/java/org/elasticsearch/cluster/coordination/NodeJoinExecutor.java

```diff
@@ -98,6 +104,7 @@ public ClusterState execute(BatchExecutionContext<JoinTask> batchExecutionContex
         }
 
         DiscoveryNodes.Builder nodesBuilder = DiscoveryNodes.builder(newState.nodes());
+        ImmutableOpenMap.Builder<String, TransportVersion> transportVersions = ImmutableOpenMap.builder(newState.transportVersions());
```

Comment:
Can we do this without ImmutableOpenMap? this is not a performance sensitive collection, it changes rarely and it doesn't need to even be ready that often if we calculate the min version once for each ClusterState.

Reply:
Changed to JDK collections